suite-native/discovery/src/discoveryMiddleware.ts

```diff
@@ -45,16 +45,25 @@ export const prepareDiscoveryMiddleware = createMiddlewareWithExtraDeps(
         // to know the device state when starting discovery of newly authorized device.
         next(action);
 
-        // On successful authorization, create discovery instance and run it.
+        // On successful authorization, create discovery instance and run it with received device state.
         if (authorizeDeviceThunk.fulfilled.match(action) && isDeviceFirmwareVersionSupported) {
             dispatch(
                 startDescriptorPreloadedDiscoveryThunk({
-                    deviceState: action.payload.state,
                     areTestnetsEnabled,
+                    forcedDeviceState: action.payload.state,
                 }),
             );
         }
 
+        // this is for further continual rerunning discovery for various other reasons
+        if (
+            deviceActions.selectDevice.match(action) ||
+            addAccountByDescriptorThunk.fulfilled.match(action) ||
```

Comment:
Same story as the other condition, `addAccountByDescriptorThunk` is not adding the account to the state so `discoveryCheckThunk` can't see the newly added account. 

```suggestion
            accountsActions.createIndexLabeledAccount.match(action) ||
```